### PR TITLE
Add the passport NFC communcation

### PR DESCRIPTION
**Pull request by GabrielFleischer**
This PR adds the "backend" for the passport NFC communication.

The UI will come in the next PR

### CODE REVIEW

File: android/app/src/main/java/com/epfl/dedis/hbt/ui/NFCViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        curCallBack?.invoke(intent)
+    }
+
+    fun setCallback(lifecycle: Lifecycle, callback: (Intent) -> Unit) {
+        if (curCallBack != null) {
+            Log.e(TAG, "A callback is already defined, cannot set a new one yet")
+            return
+        }
+
+        // Subscribe to lifecycle set and the remove callback when the observer gets destroyed
```
Comment (jbsv) [inline android/app/src/main/java/com/epfl/dedis/hbt/ui/NFCViewModel.kt:30]:
```suggestion
        // Subscribe to lifecycle set, then remove callback when the observer gets destroyed
```

File: android/app/src/main/java/com/epfl/dedis/hbt/service/passport/ncf/NFCUtils.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import com.epfl.dedis.hbt.data.Result.Error
+import com.epfl.dedis.hbt.data.Result.Success
+import com.epfl.dedis.hbt.service.passport.Passport
+import com.epfl.dedis.hbt.service.passport.mrz.MRZInfo
+import com.epfl.dedis.hbt.ui.MainActivity.Companion.getSafeParcelable
+import net.sf.scuba.smartcards.CardService
+import org.jmrtd.PassportService
+import java.security.Security
+
+object NFCUtils {
```
[the remark this answers is not in the document]
Comment (GabrielFleischer) [reply android/app/src/main/java/com/epfl/dedis/hbt/service/passport/ncf/NFCUtils.kt]: Done

File: android/app/src/main/java/com/epfl/dedis/hbt/service/passport/ncf/PassportNFC.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import java.security.GeneralSecurityException
+
+
+/**
+ * Creates a document by reading it from a service.
+ *
+ * Strongly inspired by https://github.com/jllarraz/AndroidPassportReader
+ *
+ * @param service the service to read from
+ * @param mrzInfo the BAC entries
```
[the remark this answers is not in the document]
Comment (GabrielFleischer) [reply android/app/src/main/java/com/epfl/dedis/hbt/service/passport/ncf/PassportNFC.kt]: I reworked the way MRZ and BAC interact, it should now be better

**Review by jbsv — APPROVED** [review]
LGTM, thanks

**Pull request merged**